### PR TITLE
completes task 4

### PR DESCRIPTION
**Pull request by MFStevenson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -49,4 +49,40 @@ describe("nc-news", () => {
         });
     });
   });
+
+  describe("GET /api/articles/:article_id", () => {
+    test("GET 200: returns specific article when given valid id", () => {
```
Comment (ghost) [inline __tests__/app.test.js:49]: You're expecting a specific article with the id of 1...and yet expecting any returned data. You should be expecting the correct id, correct title and so on.

File: controllers/articles-controller.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { selectArticleById } = require("../models/articles-model");
+
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+
+  selectArticleById(article_id)
+    .then((article) => {
+      if (!article) {
+        return res.status(404).send({ msg: "id not found" });
```
Comment (ghost) [inline controllers/articles-controller.js]: To be pedantic, is it the id that isn't found or the article?

File: controllers/articles-controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      } else {
+        res.status(200).send(article);
```
Comment (ghost) [inline controllers/articles-controller.js]: This should be an object with a key of article.

File: error-handler.js [hunk 4]
```diff
@@ -1,3 +1,23 @@
+exports.handleCustonErrors = (err, req, res, next) => {
```
Comment (ghost) [inline error-handler.js]: Why do you have a duplicate?

**Review by ghost — CHANGES_REQUESTED** [review]
This is almost there but some areas to tidy up. Your test needs to be more specific. You also are not testing your endpoints.json file. After lunch take a look through the comments and re submit your pull request please.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -72,17 +72,17 @@ describe("nc-news", () => {
...
-            expect(body.msg).toBe("Something wrong with input");
-          });
-      });
+      return request(app)
+        .get("/api/articles/pie")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Something wrong with input");
+        });
+    });
```
Comment (nc-mick-fay) [inline __tests__/app.test.js:86]: Nice to see you're thinking about a custom error handler for incorrect endpoints!

**Review by nc-mick-fay — APPROVED** [review]
Good actioning of feedback - merge into main when you're ready!

**Pull request merged**